File: programers_Lv1/약수의 개수와 덧셈.js
// https://school.programmers.co.kr/learn/courses/30/lessons/77884?language=javascript
function solution(left, right) {
  var answer = 0;

  const countDivisor = (number) => {
    const numberArr = Array.from({ length: number }, (_, i) => i + 1);
    const count = numberArr.reduce((acc, curr) => {
      acc += number % curr === 0 ? 1 : 0;
      return acc;
    }, 0);
    return count;
  };

  for (let i = left; i <= right; i++) {
    answer += countDivisor(i) % 2 === 0 ? i : -i;
  }

  return answer;
}

console.log(solution(13, 17));

File: programers_Lv1/약수의 개수와 덧셈.py
# https://school.programmers.co.kr/learn/courses/30/lessons/77884?language=python3
def solution(left, right):
    answer = 0

    def count_divisor(number):
        count = 0
        for i in range(1, number + 1):
            if number % i == 0:
                count += 1
        return count

    for i in range(left, right + 1):
        temp = count_divisor(i)
        if temp % 2 == 0:
            answer += i
        else:
            answer -= i

    return answer


print(solution(13, 17))
